### DISCUSSION TITLE
Loading in bubble message

### DISCUSSION THREAD

**Discussion by alahammad** [flyerhq/flutter_chat_ui#136] [Ideas & Requests]
Hello,

Thanks for great library .. I have question how to add loading view inside the bubble like show image ..

**Comment by demchenkoalex** [comment]
Hey @alahammad can you elaborate please?

**Comment by alahammad** [comment]
Hi,

I am using this library to send data and request response from server and some times to load lottie animation.

can we add loading view until request return from server or finish loading lottie ...

thanks

**Comment by demchenkoalex** [comment]
You can use custom message type to render whatever you want, since you are using Lottie you will need to use it anyway, so you can also do a loading while at it.

**Comment by alahammad** [comment]
@demchenkoalex I am trying to add loading in custom but can't, can you give example for that ..

**Comment by demchenkoalex** [comment]
@alahammad please share some code, what is exactly you're struggling with?

**Reply by alahammad** [reply]
I am trying to add loading in bubble but not clear for me. If you can provide sample.
As i checked the documentation i need to define CustomMessage but what i should send in parameter to make loading view or lottie.

**Reply by demchenkoalex** [reply]
CustomMessage has only one customisable parameter, which is metadata, which is a map. So it can contain whatever you want. And you have customMessageBuilder which gives you access to that metadata and you build whatever you want. Everything's on your side, I can't provide you a sample, because it is your project. I can only see the code (if you share something) and guide from there.

**Reply by alahammad** [reply]
I got it .. thanks
